### DISCUSSION TITLE
Need help getting NextAuth working

### DISCUSSION THREAD

**Discussion by brilang** [nextauthjs/next-auth#1721] [Help] [answered]
**Your question**

I need help getting NextAuth to work.

**What are you trying to do**

I am working on a minimal viable NextAuth Sample App for myself in order to learn how NextAuth works. I'm trying to get it working out of the box so-to-speak. I have copied code from the [Example Repo](https://github.com/nextauthjs/next-auth-example). I'm running into this error (screenshot included below copy/paste of error):

Unhandled Runtime Error
TypeError: Cannot use 'in' operator to search for 'undefined' in null

Source
.next\static\chunks\pages\_app.js (189:0) @ <unknown>

  187 |     var providers = yield getProviders();
  188 | 
> 189 |     if (!(provider in providers)) {
  190 |       window.location = "".concat(baseUrl, "/signin?callbackUrl=").concat(encodeURIComponent(callbackUrl));
  191 |       return;
  192 |     }

![image](https://user-images.githubusercontent.com/867983/114658168-36091080-9ca6-11eb-97ef-80a9e52b4917.png)

I figure I'm missing one file somewhere or a line of code I overlooked, and I need help finding what I'm missing.

**Reproduction**

[Repro Repo](https://github.com/brilang/nextexample)

In addition to the reproduction repo, I am using the following .env file. All credentials that are snipped from here have been confirmed to be correct:

```
NEXTAUTH_URL=http://localhost:3001
SECRET=<snipped>

EMAIL_SERVER=<snipped>
EMAIL_FROM=<snipped>

DATABASE_URL=<snipped>

```

My database of choice is MongoDB.

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [x] Did not find the example project helpful

**Comment by michaelcmelton** [comment, marked as answer]
@brilang You need to move your `[...nextauth].js` inside of an `auth` folder under the `api` folder. The complete path to the file should be `api/auth/[...nextauth].js`. As an aside, you should remove any providers you won't need from the [...nextauth].js file.

**Reply by balazsorban44** [reply]
good catch!

**Reply by brilang** [reply]
Nailed it! Thanks.
